test/pg_isolation_regress.sh

```diff
@@ -22,84 +28,74 @@ contains() {
     return $?
 }
 
+if [[ -z ${TEST_SCHEDULE} ]];  then
+  echo "No test schedule supplied please set ISOLATION_TEST_SCHEDULE"
+  exit 1;
+fi
+
 echo "TESTS ${TESTS}"
 echo "IGNORES ${IGNORES}"
 echo "SKIPS ${SKIPS}"
 
-if [[ -z ${TESTS} ]]; then
-    if [[ -z ${ISOLATION_TEST_SCHEDULE} ]]; then
-        for t in ${SPECS_DIR}/*.spec; do
-            t=${t##${SPECS_DIR}/}
-            t=${t%.spec}
-
-            if ! contains "${SKIPS}" "${t}"; then
-                TESTS="${TESTS} ${t}"
-            fi
-        done
-    elif [[ -n ${IGNORES} ]] || [[ -n ${SKIPS} ]]; then
-        # get the tests from the test schedule, but ignore our IGNORES
-        while read t; do
-            if [[ t =~ ignore:* ]]; then
-                t=${t##ignore:* }
-                IGNORES="${t} ${IGNORES}"
-                # run the test but ignore the result
-                TESTS="${TESTS} ${t}"
-                continue
-            fi
-            t=${t##test: }
-            ## check each individual test in test group to see if it should be ignored
-            ## note that now isolation tests are not grouped together and so the for loop
-            ## will always run once.
-            for el in ${t[@]}; do
-                if ! contains "${SKIPS}" "${el}"; then
-                    TESTS="${TESTS} ${el}"
-                fi
-            done
-        done < ${ISOLATION_TEST_SCHEDULE}
-        # no longer needed as the contents has been parsed above
-        # and unsetting it helps take a shortcut later
-        ISOLATION_TEST_SCHEDULE=
-    else
-        PG_ISOLATION_REGRESS_OPTS="${PG_ISOLATION_REGRESS_OPTS} --schedule=${ISOLATION_TEST_SCHEDULE}"
-    fi
+if [[ -z ${TESTS} ]] && [[ -z ${SKIPS} ]] && [[ -z ${IGNORES} ]]; then
+  # no filter variables set
+  # nothing to do here and we can use the cmake generated schedule
+
+  SCHEDULE=${TEST_SCHEDULE}
+
+elif [[ -z ${TESTS} ]] && [[ -z ${SKIPS} ]] && [[ -n ${IGNORES} ]]; then
+  # if we only have IGNORES we can use the cmake created schedule
+  # and just prepend ignore lines for the tests whose result should be
+  # ignored. This will allow us to retain the parallel groupings from
+  # the supplied schedule
+
+  echo > ${TEMP_SCHEDULE}
+
+  for t in ${IGNORES}; do
+      echo "ignore: ${t}" >> ${TEMP_SCHEDULE}
+  done
+  cat ${TEST_SCHEDULE} >> ${TEMP_SCHEDULE}
+
+  SCHEDULE=${TEMP_SCHEDULE}
```

Comment:
Nit: these are repeated in each branch. Put them after the if-statement?

Reply:
in the first branch the assignment is different